x/gamm/twap/hook_listener.go

```diff
@@ -35,11 +42,11 @@ func (k Keeper) GammHooks() types.GammHooks {
 
 // AfterPoolCreated is called after CreatePool
 func (hook *gammhook) AfterPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64) {
-	// err := hook.k.afterCreatePool(ctx, poolId)
-	// // Will halt pool creation
-	// if err != nil {
-	// 	panic(err)
-	// }
+	err := hook.k.afterCreatePool(ctx, poolId)
+	// Will halt pool creation
+	if err != nil {
+		panic(err)
```

Comment:
q: why are we okay with panicking here? Does this halt the chain?

Reply:
It wouldn't halt the chain, all panics are caught for txs, and should be caught in end block scopes.